### PR TITLE
feat: add html range input + tests

### LINKED ISSUE #125: Add Support for range
support for `input type=range` would be nice.

### PR DESCRIPTION
**Pull request by atymic**
This PR add the [html range input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/range).

As part of this, I updated the `BaseElement` to add a `attributeNotNull` magic method. Wasn't sure if this should be split into another PR, so let me know.

The new method was needed as `attributeIf` is a simple truthy check, which fails on `0` (which is a valid value for the `min` element.

Resolves #125

### CODE REVIEW

**Comment by atymic** [comment]
Oops, wrong button :stuck_out_tongue:

**Comment by atymic** [comment]
@sebastiandedeyne when you have a chance :)

**Comment by atymic** [comment]
@sebastiandedeyne when you have a chance :)

File: src/BaseElement.php [hunk 1]
```diff
@@ -327,6 +327,20 @@ public function unless(bool $condition, \Closure $callback)
...
+    /**
+     * Conditionally transform the element. Note that since elements are
+     * immutable, you'll need to return a new instance from the callback.
+     *
+     * @param mixed $value
+     * @param \Closure $callback
+     *
+     * @return mixed
+     */
+    public function notNull($value, \Closure $callback)
```
Comment (sebastiandedeyne) [inline src/BaseElement.php]: Could this be renamed to ìfNotNull`?

**Comment by sebastiandedeyne** [comment]
Thanks!

**Pull request merged**